### PR TITLE
Fix #533: OIDC: Configuration of mobile application

### LINKED ISSUE #533: OIDC: Configuration of mobile application
OIDC Activation Configuration

- `providerId` - identification of the configuration record, used as a key for the configuration, known by mobile application
- `clientId` - identification of the OAuth 2.0 client, required by the server and mobile application to form the URL for authorize request
- `clientSecret` - OAuth 2.0 client secret, required by the server for the token request
- `scopes` - OAuth 2.0 scopes, required by the mobile application to form the URL for authorize request
- `authorizeUri` - OAuth 2.0 authorize URI, required by the mobile application to form the URL for authorize request
- `redirectUri` - OAuth 2.0 redirect URI, required for mobile application to form the URL for authorize request
- `tokenUri` - OAuth 2.0 token URI, required by the server for the token request
- `userInfoUri` - OAuth 2.0 user info URI, required by the server for the user info request (future usage)

The parameters are configured using `pa_application_config` table as a map, the key for each record is `providerId`.

Configuration key name: `oauth2_providers`
Configuration value: 
```
[{
  "providerId": "abc123",
  "clientId": "1234567890abcdef", 
  "clientSecret": "...", 
  "scopes": "openid",
  "authorizeUri": "https://...",
  "redirectUri": "https://...",
  "tokenUri": "https://...",
  "userInfoUri": "https://...",
}, ...]
```

A new endpoint will be published using powerauth-restful-integration which will allow the mobile application to obtain OAuth 2.0 provider configuration based on its application ID and provider ID. The endpoint will convert generic PA server configuration to values required by the mobile application, stripping out sensitive values such as `clientSecret`.

URL: `POST /pa/v3/config/oidc`
ECIES scope: `APPLICATION_SCOPE`

Request:
```json
{
  "providerId": "abc123"
}
```

Response:
```json
{
  "providerId": "abc123",
  "clientId": "1234567890abcdef",
  "scopes": "openid",
  "authorizeUri": "https://...",
  "redirectUri": "https://..."
}
```

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Review by romanstrobl — APPROVED** [review]
Looks OK.

File: powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ApplicationConfigurationController.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * @param request Request OIDC application configuration.
+     * @param encryptionContext PowerAuth ECIES encryption context.
+     * @return OIDC application configuration.
+     * @throws PowerAuthApplicationConfigurationException In case there is an error while fetching claims.
+     * @throws PowerAuthEncryptionException In case of failed encryption.
+     */
+    @PowerAuthEncryption(scope = EncryptionScope.APPLICATION_SCOPE)
+    @PostMapping("oidc")
+    public OidcApplicationConfigurationResponse fetchOidcConfiguration(@EncryptedRequestBody OidcApplicationConfigurationRequest request, EncryptionContext encryptionContext) throws PowerAuthApplicationConfigurationException, PowerAuthEncryptionException {
```
Comment (banterCZ) [inline powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/controller/ApplicationConfigurationController.java:61]: @romanstrobl Don't we want to wrap it into `ObjectResponse`?

**Review by petrdvorak — CHANGES_REQUESTED** [review]
Nope, not in this repo...

File: powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/response/OidcApplicationConfigurationResponse.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * Response object for OIDC application configuration.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Data
+public class OidcApplicationConfigurationResponse {
+
+    private String providerId;
+    private String clientId;
+    private List<String> scopes;
```
Comment (banterCZ) [inline powerauth-restful-model/src/main/java/io/getlime/security/powerauth/rest/api/model/response/OidcApplicationConfigurationResponse.java:36]: Mind that in the TLDR there is
> a comma-separated array of scopes which must contain `openid`

**Comment by banterCZ** [comment]
> Nope, not in this repo...

Ok, let's move it to the enrollment server.

**Pull request closed without merging**